packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -454,7 +454,8 @@ abstract class Route<T> {
   @protected
   void dispose() {
     _navigator = null;
-    _restorationScopeId.dispose();
+    _restorationScopeId!.dispose();
+    _restorationScopeId = null;
```

Comment:
I realized this is anti-pattern to release reference inside dispose. See the updated documentation.

Reply:
In this case, the owner of the `Route` (which is really a subclass: `MaterialPageRoute`) is the `Navigator`, where it exists in a list of routes until it is popped and disposed.  There's nothing to null out on the `Navigator`, since it's stored in a list.